infra/managed-cluster/aks-cluster.bicep

```diff
@@ -0,0 +1,187 @@
+    enableOidcIssuerProfile: true
+    aadProfileAdminGroupObjectIDs: array(cluster.adminAadGroupObjectId)
+    enablePrivateCluster: true
+    usePrivateDNSZone: true
+    disableRunCommand: false
+    enablePrivateClusterPublicFQDN: false
+    aksClusterNetworkPlugin: 'azure'
+    aksClusterNetworkPluginMode: 'overlay'
+    aksClusterNetworkPolicy: 'calico'
+    aksClusterPodCidr: '172.16.0.0/16'
+    aksClusterServiceCidr: '172.18.0.0/16'
+    aksClusterDnsServiceIP: '172.18.255.250'
+    aksClusterDockerBridgeCidr: ''
+    aksClusterLoadBalancerSku: 'standard'
+    managedOutboundIPCount: 1
+    aksClusterOutboundType: 'loadBalancer'
+    aksClusterSkuTier: cluster.skuTier
+    aksClusterSshPublicKey: ''
+    aksServicePrincipalProfile: {}
+    aadProfileClientAppID: ''
+    aadProfileServerAppID: ''
+    aadProfileServerAppSecret:''
+    aadProfileTenantId: subscription().tenantId
+    primaryAgentPoolProfile: [
+      {
+        name: 'npsystem'
+        mode: 'System'
+        count: cluster.npSystem.count
+        vmSize: 'Standard_DS2_v2'
+        osDiskSizeGB: cluster.npSystem.osDiskSizeGB
+        osDiskType: 'Ephemeral'
+        osType: 'Linux'
+        osSKU: 'Ubuntu'
+        minCount: cluster.npSystem.minCount
+        maxCount: cluster.npSystem.maxCount
+        vnetSubnetId: clusterVirtualNetwork::clusterNodesSubnet.id
+        enableAutoScaling: true
+        enableCustomCATrust: false
+        enableFIPS: false
+        enableEncryptionAtHost: false
+        type: 'VirtualMachineScaleSets'
+        scaleSetPriority: 'Regular'
+        scaleSetEvictionPolicy: 'Delete'
+        orchestratorVersion: kubernetesVersion
+        enableNodePublicIP: false
+        maxPods: cluster.npSystem.maxPods
+        availabilityZones: cluster.npSystem.availabilityZones
+        upgradeSettings: {
+          maxSurge: '33%'
+        }
+        nodeTaints: [
+          'CriticalAddonsOnly=true:NoSchedule'
+        ]
+      }
+    ]
+    agentPools: [
+      {
+        name: 'npuser01'
+        mode: 'User'
+        count: cluster.npUser.count
+        vmSize: 'Standard_DS3_v2'
+        osDiskSizeGB: cluster.npUser.osDiskSizeGB
+        osDiskType: 'Ephemeral'
+        osType: 'Linux'
+        osSKU: 'Ubuntu'
+        minCount: cluster.npUser.minCount
+        maxCount: cluster.npUser.maxCount
+        vnetSubnetId: clusterVirtualNetwork::clusterNodesSubnet.id
+        enableAutoScaling: true
+        enableCustomCATrust: false
+        enableFIPS: false
+        enableEncryptionAtHost: false
+        type: 'VirtualMachineScaleSets'
+        scaleSetPriority: 'Regular'
+        scaleSetEvictionPolicy: 'Delete'
+        orchestratorVersion: kubernetesVersion
+        enableNodePublicIP: false
+        maxPods: cluster.npUser.maxPods
+        availabilityZones: cluster.npUser.availabilityZones
+        upgradeSettings: {
+          maxSurge: '33%'
+        }
+      }
+    ]
+    autoScalerProfileBalanceSimilarNodeGroups: 'false'
```

Comment:
Is the default value of all of the following parameters different than what we are changing to? if same then probably we should remove them from here.

Reply:
Yeah, they are default values in the module, but I want to specify them explicitly because we will have visibility on the configuration.  Also, will prevent any wanted changes when you upgrade the module version.

Reply:
i agree with this approach Praveen